src/Toolkit/Collection.php

```diff
@@ -936,12 +936,33 @@ public function sortBy()
         // build the multisort params in the right order
         $params = [];
 
+        // duplicate values set false as default
+        $hasDuplicateValues = false;
+
         foreach ($fields as $field) {
-            $params[] = $field['values']    ?? [];
+            $values = [];
+            if (empty($field['values']) === false) {
+                $values = array_filter($field['values']) ? array_values($field['values']) : array_keys($field['values']);
```

Comment:
This is for stable work. For ex: in files collection array values are empty, but pages collection array keys and values are not empty. When the values are empty, it makes sort wrong because of the new feature we added as numeric sorting.